tests/unit/GooglePlacesUtilsTest.js

```diff
@@ -0,0 +1,142 @@
+import {getAddressComponent, validateAddressComponents} from '../../src/libs/GooglePlacesUtils';
+
+describe('GooglePlacesUtilsTest', () => {
+    describe('validateAddressComponents', () => {
+        it('should reject Google Places result with missing street number', () => {
+            // This result appears when searching for "25220 Quail Ridge Road, Escondido, CA, 97027"
+            const googlePlacesRouteResult = {
+                address_components: [
+                    {
+                        long_name: 'Quail Ridge Road',
+                        short_name: 'Quail Ridge Rd',
+                        types: ['route'],
+                    },
+                    {
+                        long_name: 'Escondido',
+                        short_name: 'Escondido',
+                        types: ['locality', 'political'],
+                    },
+                    {
+                        long_name: 'San Diego County',
+                        short_name: 'San Diego County',
+                        types: ['administrative_area_level_2', 'political'],
+                    },
+                    {
+                        long_name: 'California',
+                        short_name: 'CA',
+                        types: ['administrative_area_level_1', 'political'],
+                    },
+                    {
+                        long_name: 'United States',
+                        short_name: 'US',
+                        types: ['country', 'political'],
+                    },
+                    {
+                        long_name: '92027',
+                        short_name: '92027',
+                        types: ['postal_code'],
+                    },
+                ],
+                formatted_address: 'Quail Ridge Rd, Escondido, CA 92027, USA',
```

Comment:
I left this here even if it not used in the tests just as a way to remember these addresses that gave up problems.